src/components/Advertiser/AdvertiserTable.vue

```diff
@@ -644,10 +650,17 @@ async function onUpdate(e) {
     label: e.label
   }
 }
-
-watch(selectedDataType, (newType) => {
```

Comment:
So this watch is responsible for fetching only the selected advertises instead of fetching them all and then just filter them out by status. This is needed because of amount of requests are made to firestore. So basically our main goal is to optimize the app that way so we make as few requests as possible (make them only where its necessary) 

You can see the amount of requests being made to fetch all advertises VS to fetch only 1. Thats still a lot of requests even 1 advertisement, but it's less than to fetch all.

Imaginge we will have even more than that for all ads. 

https://github.com/user-attachments/assets/91743ceb-3300-4ccd-ab63-023fbdd904d9

https://github.com/user-attachments/assets/201804dd-04c0-4409-b650-6ff38578dab7

Reply:
These requests are being made in order to compute the total number of visits, impressions, and clicks for each advertisement.
Rather than recalculating each time we fetch an advertisement, we can update the cloud function to directly update the advertisement document with the total impressions, total clicks, and total visits. So that we don't need to fetch the impressions, clicks, or visits collection to compute total data for each advertisement.